src/Database/PostgreSQL/Simple/FromField.hs

```diff
@@ -487,6 +487,15 @@ instance FromField LocalTimestamp where
 instance FromField Date where
   fromField = ff TI.dateOid "Date" parseDate
 
+-- | interval. Requires you to configure intervalstyle as @iso_8601@.
+--
+--   You can configure intervalstyle on every connection with a @SET@ command,
+--   but for better performance you may want to configure it permanently in the
+--   file found with @SHOW config_file;@ .
+--
```

Comment:
See https://www.postgresql.org/docs/12/datatype-datetime.html#DATATYPE-INTERVAL-OUTPUT for documentation on different interval output formats.

This was an unexpected stumbling block for me, it seems you have to ask Postgres to give you ISO-compliant `interval`s, with the default alternative being a vendor-specific `postgres` format. Searching for `intervalstyle` reveals that it's quite popular for developers to want to set this to something ISO8601-compliant, so hopefully this won't detract too much from adoption.

The long-term alternative is probably to support the four possible `intervalstyle`s available, so that the user doesn't need to worry about this problem.

Also, this may be a latent issue with `UTCTime` as well. `timestamptz` also seems to offer multiple output formats, and I got the sense from browsing the parsers that this library only supports the default one. Check https://www.postgresql.org/docs/12/datatype-datetime.html#DATATYPE-DATETIME-OUTPUT .